### PR TITLE
Patch niminst (de)install to be usable for distros

### PR DESCRIPTION
**Pull request by kasperk81**
* Add support for DESTDIR variable (staged installs).
* Change paths to be FHS/Alpine compliant.
<del>* Use "install" command instead of mkdir/cp/chmod.</del>

cc @jirutka

### CODE REVIEW

**Comment by Araq** [comment]
> Use "install" command instead of mkdir/cp/chmod.

The install command is not part of Posix.

File: config/nim.cfg [hunk 1]
```diff
@@ -44,7 +44,7 @@ path="$lib/core"
 path="$lib/pure"
 
 @if not windows:
-  nimblepath="/opt/nimble/pkgs/"
```
Comment (Araq) [inline config/nim.cfg:47]: This is a breaking change. Not sure how to support "/usr/share"

File: compiler/modulepaths.nim [hunk 2]
```diff
@@ -87,6 +87,10 @@ when false:
           result = findInNimbleDir(pkg, subdir, nimbleDir / "pkgs")
           if result.len > 0: return result
           when not defined(windows):
+            result = findInNimbleDir(pkg, subdir, "/usr/share/nimble/pkgs")
+            if result.len > 0: return result
+            result = findInNimbleDir(pkg, subdir, "/usr/local/share/nimble/pkgs")
```
Comment (Araq) [inline compiler/modulepaths.nim:92]: But what about "usr/share/local"?

Comment (kasperk81) [reply compiler/modulepaths.nim:92]: We currently use /usr/share or /usr/local/share in install and deinstall scripts, so I added probes for knows as known locations. Do you want me to add /usr/share/local as well in those scripts and here?

Comment (Araq) [reply compiler/modulepaths.nim:92]: Sorry I have no idea if "usr/share/local" exists XD, it was a bad joke.

File: config/nim.cfg (hunk 1 above)
Comment (kasperk81) [reply config/nim.cfg:47]: This PR is updating all paths for [FHS](https://www.pathname.com/fhs/pub/fhs-2.3.html#PURPOSE18) compliance in one go, so it won't break it. Even today, if we pass /usr/bin to install.nimf, this configuration needs to be updated manually after installation.

**Pull request closed without merging**